pypeit/core/gui/skysub_regions.py

```diff
@@ -0,0 +1,552 @@
+"""
+This script allows the user to manually select the sky background regions
+"""
+
+import os, sys
+import copy
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.widgets import Button, Slider
+import matplotlib.transforms as mtransforms
+from scipy.interpolate import RectBivariateSpline
+
+from pypeit import specobjs
+from pypeit import msgs
+
+operations = dict({'cursor': "Select object trace (LMB click)\n" +
+                   "         Navigate (LMB drag = pan, RMB drag = zoom)\n" +
+                   "         Note: In order to pan/zoom, you need to first activate\n" +
+                   "         the pan/zoom tool with the 'p' key, or by selecting the\n" +
+                   "         pan/zoom tool on the Matplotlib navigation tool menu. You\n" +
+                   "         can also zoom using the magnifying glass (select this option\n" +
+                   "         from the Matplotlib navigation tool menu)\n",
+                   'a': "Add a new object trace using the selected method",
+                   'c': "Center the window at the location of the mouse",
+                   'd': "Delete selected object trace",
+                   'h/r': "Return zoom to the original plotting limits",
+                   'p': "Toggle pan/zoom with the cursor",
+                   '?': "Display the available options",
+                   })
+
+
+class SkySubGUI(object):
+    """
+    GUI to interactively define the sky regions. The GUI can be run within
+    PypeIt during data reduction, or as a standalone script outside of
+    PypeIt. To initialise the GUI, call the initialise() function in this
+    file.
```

Comment:
Please make it more clear in the docs what this GUI produces (an output file, a sky-subtracted frame?), and what the control flow is to get from running it to sky-subtraction.

Reply:
OK - after discussing on Slack, I'll proceed with outputting a fits file (which we can set in the `.pypeit` file... somehow). I'll start implementing this approach and the relevant documentation... stay tuned.